azkaban-exec-server/src/main/java/azkaban/execapp/ExecutorServlet.java

```diff
@@ -357,16 +359,42 @@ private void handleReloadJobTypePlugins(Map<String, Object> respMap)
   private void setActive(boolean value, Map<String, Object> respMap)
       throws ServletException {
     try {
-      ExecutorLoader executorLoader = application.getExecutorLoader();
-      Executor executor = executorLoader.fetchExecutor(application.getHost(), application.getPort());
-      Preconditions.checkState(executor != null, "Unable to obtain self entry in DB");
-      if (executor.isActive() != value) {
-        executor.setActive(value);
-        executorLoader.updateExecutor(executor);
-        flowRunnerManager.setActive(value);
-      } else {
-        logger.warn("Set active action ignored. Executor is already " + (value? "active" : "inactive"));
-      }
+      setActiveInternal(value);
+      respMap.put(STATUS_PARAM, RESPONSE_SUCCESS);
+    } catch (Exception e) {
+      logger.error(e);
+      respMap.put(RESPONSE_ERROR, e.getMessage());
+    }
+  }
+
+  private void setActiveInternal(boolean value)
+      throws ExecutorManagerException {
+    ExecutorLoader executorLoader = application.getExecutorLoader();
+    Executor executor = executorLoader.fetchExecutor(application.getHost(), application.getPort());
+    Preconditions.checkState(executor != null, "Unable to obtain self entry in DB");
+    if (executor.isActive() != value) {
+      executor.setActive(value);
+      executorLoader.updateExecutor(executor);
+      flowRunnerManager.setActive(value);
+    } else {
+      logger.warn("Set active action ignored. Executor is already " + (value? "active" : "inactive"));
+    }
+  }
+
+  /**
+   * Prepare the executor for shutdown.
+   *
+   * @param respMap json response object
+   * @throws ServletException
+   */
+  private void shutdown(Map<String, Object> respMap)
+      throws ServletException {
+    try {
+      logger.warn("Shutting down executor...");
+
+      // Set the executor to inactive. Will receive no new flows.
+      setActiveInternal(false);
```

Comment:
No. The shutdown would fail. It will be unsafe to shutdown the executor while it might still receive active traffic. right ? The goal is to set the executor to inactive first before attempting shutdown.

If you want to shutdown now, you can just use the shutdown bash script or kill the process.